### ISSUE TITLE
#eval consumes SIGINT

### ISSUE THREAD

**Issue by jbodah** [clbustos/rinruby#48]
I noticed a weird issue with our test suite where sending SIGINT would sometimes cause my main test process to exit but my tests would keep going (sometimes it just keeps going and ignores any SIGINT)

We're using `Rake::TestTask` which will shell out a new Ruby process. The problem I found was that if we call `rinruby`'s `#eval` function then it will setup SIGINT handlers and cause the test process not to exit. I feel like this behavior should be flaggable

I noticed this with rinruby 2.1.0 but get the same behavior on master

**Comment by fenrir-naru** [comment]
@jbodah I'm a contributor. Thank you for your report. I agree with you to revise the signal handling to interrupt the eval loop smarter.

**Comment by fenrir-naru** [comment]
@jbodah I have revised the signal handling. Try my [better_eval_sigint_handling branch](https://github.com/fenrir-naru/rinruby/tree/better_eval_sigint_handling) in a forked version.
```shell
gem install specific_install # Only required first time
gem specific_install https://github.com/fenrir-naru/rinruby.git better_eval_sigint_handling
```
is useful.

**Comment by fenrir-naru** [comment]
@clbustos This issue may be solved by merging #49. Please close it after merge of #49. In addition, please bump version.
@jbodah If you have further problem, please reopen it to let me know.
